File: better_player/android/src/main/java/com/jhomlala/better_player/BetterPlayerCache.java
package com.jhomlala.better_player;

import android.content.Context;
import android.util.Log;

import com.google.android.exoplayer2.database.ExoDatabaseProvider;
import com.google.android.exoplayer2.upstream.cache.LeastRecentlyUsedCacheEvictor;
import com.google.android.exoplayer2.upstream.cache.SimpleCache;

import java.io.File;

public class BetterPlayerCache {
    private static volatile SimpleCache instance;

    public static SimpleCache createCache(Context context, long cacheFileSize) {
        if (instance == null) {
            synchronized (BetterPlayerCache.class) {
                if (instance == null) {
                    instance = new SimpleCache(
                            new File(context.getCacheDir(), "betterPlayerCache"),
                            new LeastRecentlyUsedCacheEvictor(cacheFileSize),
                            new ExoDatabaseProvider(context));
                }
            }
        }
        return instance;
    }

    public static void releaseCache() {
        try {
            if (instance != null) {
                instance.release();
                instance = null;
            }
        } catch (Exception exception) {
            Log.e("BetterPlayerCache", exception.toString());
        }
    }
}

File: packages/better_player_android/android/src/main/kotlin/pl/hasoft/better_player/BetterPlayerCache.kt
package pl.hasoft.better_player

import android.content.Context
import android.util.Log
import androidx.annotation.OptIn
import androidx.media3.common.util.UnstableApi
import androidx.media3.datasource.cache.SimpleCache
import androidx.media3.datasource.cache.LeastRecentlyUsedCacheEvictor
import androidx.media3.database.ExoDatabaseProvider
import java.io.File
import java.lang.Exception

@OptIn(UnstableApi::class)
object BetterPlayerCache {
    @Volatile
    private var instance: SimpleCache? = null
    fun createCache(context: Context, cacheFileSize: Long): SimpleCache? {
        if (instance == null) {
            synchronized(BetterPlayerCache::class.java) {
                if (instance == null) {
                    instance = SimpleCache(
                        File(context.cacheDir, "betterPlayerCache"),
                        LeastRecentlyUsedCacheEvictor(cacheFileSize),
                        ExoDatabaseProvider(context)
                    )
                }
            }
        }
        return instance
    }

    @JvmStatic
    fun releaseCache() {
        try {
            if (instance != null) {
                instance!!.release()
                instance = null
            }
        } catch (exception: Exception) {
        }
    }
}